packages/snaps-execution-environments/src/common/BaseSnapExecutor.test.browser.ts

```diff
@@ -1504,6 +1505,43 @@ describe('BaseSnapExecutor', () => {
     });
   });
 
+  it('supports onUserInput export', async () => {
+    const CODE = `
+      module.exports.onUserInput = ({ id, event }) => {}
+    `;
+
+    const executor = new TestSnapExecutor();
+    await executor.executeSnap(1, MOCK_SNAP_ID, CODE, []);
+
+    expect(await executor.readCommand()).toStrictEqual({
+      jsonrpc: '2.0',
+      id: 1,
+      result: 'OK',
+    });
+
+    const params = {
+      id: 'foo',
+      event: {
+        type: UserInputEventTypes.ButtonClickEvent,
+        name: 'bar',
+      },
+    };
+
+    await expect(
+      executor.writeCommand({
+        jsonrpc: '2.0',
+        id: 2,
+        method: 'snapRpc',
+        params: [
+          MOCK_SNAP_ID,
+          HandlerType.OnUserInput,
+          MOCK_ORIGIN,
+          { jsonrpc: '2.0', method: 'foo', params },
+        ],
+      }),
+    ).resolves.not.toThrow();
```

Comment:
This won't necessarily throw, we should read the response and assert no errors found